### DISCUSSION TITLE
The path specification method for test.include and coverage.include is different.

### DISCUSSION THREAD

**Discussion by yuta-nishi** [vitest-dev/vitest#5232] [General]
My `vitest.config.ts` is as follows. Without any configuration for coverage, files under `node_modules` were also displayed, so I added a configuration for `coverage.include`. However, it didn't work well when I set it up in the same way as `test.include`, and it took me some time.

Why do the specification methods differ? I feel it would be more understandable if they worked with the same configuration.

```typescript
// vitest.config.ts

import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vitest/config';

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    include: ['./src/tests/**/*.test.{ts,tsx}'],
    globals: true,
    environment: 'happy-dom',
    setupFiles: './vitest.setup.ts',
    coverage: {
      // Note that the path specification method for test.include is different. The following will not work.
      // include: ['./src/**/*.ts', 'src/**/*.{ts,tsx}'],
      include: ['src/**/*.ts', 'src/**/*.tsx'],
      exclude: ['src/components/ui'],
    },
  },
});
```

**Comment by AriPerkkio** [comment]
> Without any configuration for coverage, files under `node_modules` were also displayed

This definitely sounds like a bug. By default Vitest coverage should exclude all `node_modules` as configured below. Please file a bug report with reproducible setup and let's fix this.

https://github.com/vitest-dev/vitest/blob/e191e658d5052cb335ffe163a045ed294238d341/packages/coverage-v8/src/provider.ts#L99

> Why do the specification methods differ? I feel it would be more understandable if they worked with the same configuration.

Maybe the underlying dependencies work differently. Could you try if `coverage.allowExternal: true` makes any difference?

**Comment by yuta-nishi** [comment]
>  Please file a bug report with reproducible setup and let's fix this.

I'm very sorry for any inconvenience. Despite reproducing the code from scratch and making improvements to the existing code, I could not replicate the bug where coverage was targeting the entire range.

As a solution, deleting the entire project and then running `git clone` resolved the issue. However, deleting generated folders such as `node_modules` or `build` and then running `pnpm install` or `pnpm dev` again did not resolve the issue.

Repeatedly running `pnpm update` locally may have been the cause, but I am not certain. For reference, I have attached a part of the coverage results.

- Normal

```
% Coverage report from v8
----------------------------|---------|----------|---------|---------|-----------------------------
File                        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------------|---------|----------|---------|---------|-----------------------------
All files                   |       4 |    29.41 |   13.33 |       4 |
 url-copy                   |       0 |        0 |       0 |       0 |
  postcss.config.js         |       0 |        0 |       0 |       0 | 1-9
  tailwind.config.js        |       0 |        0 |       0 |       0 | 1-71
 url-copy/src/background    |   17.36 |    66.66 |      50 |   17.36 |
```

- Abnormal

```
% Coverage report from v8
-------------------------------------------------|---------|----------|---------|---------|-------------------
File                                             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------------------------------------|---------|----------|---------|---------|-------------------
All files                                        |   98.52 |     4.16 |    4.16 |   98.52 |
 extension-template                              |       0 |        0 |       0 |       0 |
  postcss.config.js                              |       0 |        0 |       0 |       0 | 1-9
  tailwind.config.js                             |       0 |        0 |       0 |       0 | 1-71
 ...template/build/chrome-mv3-dev/.plasmo/static |     100 |      100 |     100 |     100 |
  content.tsx                                    |     100 |      100 |     100 |     100 |
  devtools.tsx                                   |     100 |      100 |     100 |     100 |
.
.
.
extension-template/src/components               |       0 |        0 |       0 |       0 |
  CountButton.tsx                                |       0 |        0 |       0 |       0 | 1-18
 extension-template/src/contents                 |   17.24 |       50 |      50 |   17.24 |
  hello.tsx                                      |     100 |      100 |     100 |     100 |

```

> Maybe the underlying dependencies work differently

`coverage.include` was compatible with `{ts,tsx}` but not with `./`.
